repose-aggregator/src/docs/asciidoc/recipes/docker.adoc

```diff
@@ -1,130 +1,147 @@
+    The volume mapping format is `HOST_DIRECTORY:CONTAINER_DIRECTORY`.
+    So in this case, the contents of `/my/config/directory` on the host will be visible within in the container in the `/etc/repose` directory.
+    This allows us to modify configuration files without having to rebuild the Docker image or even restart the Docker container!
 <4> This option defines a port mapping between the host (i.e., the OS running Docker) and the Docker container.
-    Doing so allows Repose to accept traffic over the specified port from outside of Docker.
-    The port mapping format is HOST_PORT:CONTAINER_PORT, so in this case, traffic on port 8080 of the host will be forwarded to port 8080 of the container.
+    Doing so allows *Repose* to accept traffic over the specified port from outside of Docker.
+    The port mapping format is `HOST_PORT:CONTAINER_PORT`.
+    So in this case, traffic on port `8080` of the host will be forwarded to port `8080` of the container.
 <5> This option sets the `JAVA_OPTS` environment variable.
-    Doing so allows the user to configure the JVM that Repose is running on.
+    Doing so allows the user to configure the JVM that *Repose* is running on.
     The value of this environment variable should be valid command-line arguments for the `java` command.
 <6> This option names the container so that it's easier to interact with.
-<7> This is the image ID (in repository:tag format) to create a container from.
-    Remember that Repose images are hosted at Docker Hub, which Docker can use implicitly.
+<7> This is the Image ID (in `repository:tag` format) to create a container from.
+    Remember that *Repose* images are hosted at Docker Hub, which Docker can use implicitly.
+    Otherwise use the `repository:tag` of the image you created.
 
-Now if we wanted to, we could change the Docker options that define the environment that Repose will run in.
-If we wanted to forward the random ports instead of explicitly declaring the port mapping, we could drop the `-p 8080:8080` option and pass the `-P` flag instead.
+Now if we wanted to, we could change the Docker options that define the environment that *Repose* will run in.
+If we wanted to forward random ports instead of explicitly declaring the port mapping, we could replace the `--publish 8080:8080` option with `--publish-all` instead.
 
-After running that command, you should be able to see Repose running inside a Docker container!
-Use the following command to show all running containers:
+== Manually Building a Repose Docker Image
+The latest version of the Dockerfiles used to build the *Repose Docker* images published on Docker Hub can be found at:
 
-----
-docker ps
-----
+- https://raw.githubusercontent.com/rackerlabs/repose/master/repose-aggregator/artifacts/docker/src/docker/ubuntu/Dockerfile[Debian/Ubuntu]
+- https://raw.githubusercontent.com/rackerlabs/repose/master/repose-aggregator/artifacts/docker/src/docker/centos/Dockerfile[RedHat/Centos]
```

Comment:
Why not leverage the fact that our docs are versioned? You can link to the actual Dockerfiles in the repository, though if you link rather than include, you will probably have to copy those files into the docs project (like we do with schemas).

Reply:
That is kind of where I was at. I didn't feel the need to `include::` it in, but at the same time wanted to make it as accessible as possible.
I didn't like this solution either, but couldn't think of a better one.

Do we have access to the version number in a way we could reference it during the doc build?

Reply:
Sure, the Asciidoctor plugin for Gradle sets the `project-version` attribute.

Source: http://asciidoctor.org/docs/asciidoctor-gradle-plugin/#options-attributes

I would still probably suggest just pulling the files you care about over just so that resources are stored and served consistently. That is, resources are served from the same servers and domain. Actually, I'm not sure if that's valid since we are hosting on GitHub anyway. I'm not sure if GitHub pages content is distributed the same was as Git repository, and while I would hope that GitHub pages do not go down every time GitHub.com does, I couldn't say for certain. Even so, pulling the files in insulates us from that concern if we do every switch hosts for some reason.

Reply:
Ok. So these were brought in as includes and then wrapped with the HTML to make them expandable.

Thoughts?